py/jupyterlite/src/jupyterlite/addons/federated_extensions.py

```diff
@@ -122,137 +127,112 @@ def resolve_one_extension(self, path_or_url, init):
-            actions=[_extract],
+        yield dict(
+            name=f"copy:{archive.name}",
+            file_dep=[hashfile],
+            actions=[(self.copy_all_federated_extensions, [unarchived])],
         )
 
+    def copy_all_federated_extensions(self, unarchived):
+        """actually copy all federated extensions found in a folder."""
+        for simple_pkg_json in unarchived.rglob(
+            f"{SHARE_LABEXTENSIONS}/*/package.json"
+        ):
+            self.copy_one_federated_extension(simple_pkg_json)
+        for org_pkg_json in unarchived.rglob(
+            f"{SHARE_LABEXTENSIONS}/@*/*/package.json"
+        ):
+            self.copy_one_federated_extension(org_pkg_json)
+
+    def copy_one_federated_extension(self, pkg_json):
+        """actually copy one labextension from an extracted archive"""
+        pkg_data = json.loads(pkg_json.read_text(**UTF8))
+
+        if self.is_prebuilt(pkg_data):
+            pkg_name = pkg_data["name"]
+            self.copy_one(pkg_json.parent, self.output_extensions / f"{pkg_name}")
+
     def is_prebuilt(self, pkg_json):
         """verify this is an actual pre-built extension, containing load information"""
         return pkg_json.get("jupyterlab", {}).get("_build", {}).get("load") is not None
 
-    def copy_conda_extensions(self, conda_pkg):
-        """copy the labextensions from a local conda package"""
-        import tarfile
-
-        with tarfile.open(conda_pkg, "r:bz2") as zf:
-            infos = [*zf.getmembers()]
-
-            for info in infos:
-                filename = info.name
-                if filename.startswith(SHARE_LABEXTENSIONS) and filename.endswith(
-                    PACKAGE_JSON
-                ):
-                    is_prebuilt = False
-                    try:
-                        is_prebuilt = self.is_prebuilt(
-                            json.loads(zf.extractfile(info).read().decode("utf-8"))
-                        )
-                    except Exception as err:
-                        print(f"... skipping {info}: {err}")
-
-                    if is_prebuilt:
-                        yield from self.extract_one_conda_extension(
-                            conda_pkg, info, infos
-                        )
-
-    def extract_one_conda_extension(self, conda_pkg, pkg_json_info, all_infos):
-        """extract one labextension from a conda package"""
-        import tarfile
-
-        pkg_root_with_slash = pkg_json_info.name.split(PACKAGE_JSON)[0]
-        prefix = len(pkg_root_with_slash)
-        stem = pkg_root_with_slash.split(SHARE_LABEXTENSIONS)[1][1:-1]
-        dest = self.output_extensions / stem
-
-        def _filter_members(infos):
-            return [
-                p
-                for p in infos
-                if not p.isdir() and p.name.startswith(pkg_root_with_slash)
-            ]
-
-        targets = [dest / m.name[prefix:] for m in _filter_members(all_infos)]
+    def copy_conda2_extensions(self, conda_pkg):
```

Comment:
Should this method still be called `copy_conda_extensions`? Or does the `2` refer to something in particular?

Reply:
Yeah, it's referencing that fact that under the hood, these are built (currently) only when you specify [`pkg_format: 2`](https://conda.io/projects/conda/en/latest/user-guide/configuration/use-condarc.html#specify-conda-build-package-version-pkg-version)

Reply:
ok thanks :+1: